cpp/src/arrow/array-decimal-test.cc

```diff
@@ -75,8 +75,7 @@ class DecimalTest : public DecimalTestBase<T> {
  public:
   std::vector<uint8_t> data(
       const std::vector<T>& input, size_t byte_width) const override {
-    std::vector<uint8_t> result;
-    result.reserve(input.size() * byte_width);
+    std::vector<uint8_t> result(input.size() * byte_width);
```

Comment:
Interesting, I wonder why this wasn't accessing out of bounds and aborting the program. I wonder if it ever gets called.